drivers/wifi/wfx200/wfx200_host.c

```diff
@@ -0,0 +1,730 @@
+/**
+ * Copyright (c) 2022 grandcentrix GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME wifi_wfx200_host
+#define LOG_LEVEL CONFIG_WIFI_LOG_LEVEL
+
+#include <sl_wfx.h>
+#undef BIT
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <stdarg.h>
+#include <stdint.h>
+
+#include <zephyr.h>
+#include <device.h>
+#include <sys/util.h>
+#include <drivers/spi.h>
+#include <drivers/gpio.h>
+
+#include "wfx200_internal.h"
+
+#include <sl_wfx_wf200_C0.h>
+
+/* The driver from SiLabs does use a internal global variable to store the
+ * drivers context. This means, that we are limitied to one instance of SiLabs
+ * driver and need to use a global variable here as well, as the host api does
+ * not give a context to the callbacks. So we need to use our own global
+ * variable here as well.
+ */
+extern struct wfx200_data *wfx200_0;
+
+/* WFX host callbacks */
+void sl_wfx_connect_callback(sl_wfx_connect_ind_t *connect);
+void sl_wfx_disconnect_callback(sl_wfx_disconnect_ind_t *disconnect);
+void sl_wfx_start_ap_callback(sl_wfx_start_ap_ind_t *start_ap);
+void sl_wfx_stop_ap_callback(sl_wfx_stop_ap_ind_t *stop_ap);
+void sl_wfx_host_received_frame_callback(sl_wfx_received_ind_t *rx_buffer);
+void sl_wfx_scan_result_callback(sl_wfx_scan_result_ind_t *scan_result);
+void sl_wfx_scan_complete_callback(sl_wfx_scan_complete_ind_t *scan_complete);
+void sl_wfx_generic_status_callback(sl_wfx_generic_ind_t *frame);
+void sl_wfx_ap_client_connected_callback(sl_wfx_ap_client_connected_ind_t *ap_client_connected);
+void sl_wfx_ap_client_rejected_callback(sl_wfx_ap_client_rejected_ind_t *ap_client_rejected);
+void sl_wfx_ap_client_disconnected_callback(sl_wfx_ap_client_disconnected_ind_t *ap_client_disconnected);
+void sl_wfx_ext_auth_callback(sl_wfx_ext_auth_ind_t *ext_auth_indication);
+
+/**
+ * Functions for initializing the host, firmware uploading and pds
+ */
+sl_status_t sl_wfx_host_init(void)
+{
+	/* Initialization happens in wfx200_init, nothing to do here */
```

Comment:
It would make sense to expose this to the application, because various applications might require additional initialization steps.

Reply:
I do not agree, that it is necessary for an application to hook onto the wifi driver initialization. This is the device driver, we intent to abstract this away from the application developer. The application might be interested on a "got a ip address" event, but not "some driver init did happen".

Reply:
We can't know the requirements of the application in advance.
Obviously, I agree that it doesn't have to be necessary. 
As other callbacks, it should be optional to be able to subscribe for it.